### PR TITLE
Added basic setup and database structure

### LINKED ISSUE #1: Database Structure
```
    +-------------+ 1   n +-------------+
    |   Indexes   |-------|  Mappings   |
    +------+------+       +------+-------+
           | 1                   | 1
           |                     |
           |                     |
           |                     |
           | n                   | n
    +------+------+ 1   n +------+------+ 1   n +-------------+ n   1 +-------------+
    |  Documents  |-------|   Fields    |-------|   Tokens    |-------|    Terms    |
    +-------------+       +-------------+       +-------------+       +-------------+
```

**Indexes**

 - name (string)

**Documents**

 - id (string)
 - type (string)
 - document (text)
 - index_id (string)

**Mappings**

 - id (integer)
 - name(string)
 - analyzed(boolean)?
 - analyzer(string)?

**Fields**

 - id (integer)
 - document_id (string)
 - name (string)
 - value (text)?

**Token**

 - id (integer)
 - token_id (integer)
 - field_id (integer)
 - position (string)

**Term**

 - id (integer)
 - value (string) (unique)

**TODO**

 - [ ] Table encoding?

### PR DESCRIPTION
**Pull request by alexander-schranz**
Added basic database structure.

fixes #1

### CODE REVIEW

File: src/Entity/Document.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+namespace Pucene\Bundle\DoctrineBundle\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+
+/**
+ * Document entity.
+ */
+class Document
```
Comment (wachterjohannes) [inline src/Entity/Document.php:11]: This classes should be placed in the library and in the bundle only the doctrin-xml should be placed.

`Pucene\Model\Document`

what do you think about this idea?

File: src/Entity/Token.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Pucene\Bundle\DoctrineBundle\Entity;
+
+/**
+ * Token entity.
+ */
+class Token
```
Comment (wachterjohannes) [inline src/Entity/Token.php:8]: We need another entity - else we would duplicate a lot the tokens

```
Token        ->     TokenField          ->    Field
+ value             + pos/start/end
```

Comment (alexander-schranz) [reply src/Entity/Token.php:8]: as said we need todo a benchmark if its faster to have an additional entity or store the string here directly with an index.

File: src/Entity/Document.php (hunk 1 above)
Comment (alexander-schranz) [reply src/Entity/Document.php:11]: @wachterjohannes for me definitly not as this class is not needed to be in the core. 

For me the architecture should be the following:

There is a the pucene bundle which implements query builder, read type configuration from xml, yml or annotation, mapping and analyze. There is a simple config parameter `storage` which you can use the `raw`, `elasticsearch`, `mysql`. 

The storage bundle itself need to implement the StorageInterface.

```
public function request($json, $method)
```

The core or elasticsearch storage dont need any Index, Document, Mapping, Field or Token entities with relation to each others they are just needed for the doctrine implemention to make queries.

File: src/Entity/Token.php (hunk 2 above)
Comment (wachterjohannes) [reply src/Entity/Token.php:8]: ok thats clear (: but i think we will have a size problem before we will have any performance problem!

File: src/Entity/Document.php (hunk 1 above)
Comment (wachterjohannes) [reply src/Entity/Document.php:11]: hopefully it will work (: we will see!

**Pull request merged**